### PR TITLE
fatfs: Add RxDOS.2 (RxDOS v7.23+) boot protocol support

### PR DESCRIPTION
**Pull request by ecm-pushbx**
Previously requested and described in this thread:
https://github.com/stsp/dosemu2/issues/581#issuecomment-373933904

Tested with "RxDOS version 7.24 [hg 8b75542e6ba1] [2018-04-26]", from
https://bitbucket.org/ecm/rxdos-7.2x/ (+ lmacros + ldos + ldosboot)

This changes NEWPCD_D to (PC_D | (1 << 30)) because MAX_SYS_IDX is
now 23.

### CODE REVIEW

**Comment by ecm-pushbx** [comment]
Here are the files I used to boot: [rxdos-8b75542e6ba1-test-load.zip](https://github.com/stsp/dosemu2/files/1951721/rxdos-8b75542e6ba1-test-load.zip), built using `$ use_build_revision_id=1 ./mak.sh -D_DEBUG4`

After "Test payload loaded." it takes a key press to continue booting. With umb.sys in RXCONFIG.SYS, the load should relocate DOSCODE to the HMA. Note that the "1st name" (and subsequent "Boot drive and file name") is "RXDOS.COM", which is parsed from the name specified in blank-padded form in the boot sector.

File: src/base/misc/fatfs.h [hunk 1]
```diff
@@ -69,6 +71,14 @@ typedef struct {
 
 } fatfs_t;
 
+struct rxdos_loadstackvars {
```
Comment (stsp) [inline src/base/misc/fatfs.h]: Please move this struct to where you use it.
It is not needed in a public header.

Comment (ecm-pushbx) [reply src/base/misc/fatfs.h]: Should it go near the top, or directly in front of the function that uses it?

Comment (ecm-pushbx) [reply src/base/misc/fatfs.h]: But with that option, how would I specify the cast to that pointer's type?

Comment (andrewbird) [reply src/base/misc/fatfs.h]: Do you need to cast, LINEAR2UNIX() returns a void * ?

Also sizeof(struct rxdos_loadstackvars) could be sizeof(*lsv)

**Comment by ecm-pushbx** [comment]
Oh allright. I'll update the patch later today.

**Comment by ecm-pushbx** [comment]
Updated! Put the structure definition into the case's block as suggested. Also added a check for minimum size of the file.

**Pull request merged**